feat: Split legend interaction (click vs label) with hover effects

joepio opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


https://github.com/user-attachments/assets/177224cf-cff1-4cd6-853e-f435ab5a24d0



## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->


This PR refines the behavior of the legend in the **Statistics Graph** and **History Graph** cards.
Current behavior: clicking label or colored bullet toggles chart visibility. Proposed change:

- **Clicking the colored bullet**: Toggles the visibility of the data series (standard behavior).
- **Clicking the label**: Opens the "more-info" dialog for the corresponding entity.

**UI Enhancements**:
- Added separate tooltips for the bullet ("Toggle visibility") and the label ("Show more info").
- Added visual hover hints: bullet opacity changes (50%), and label is underlined to indicate interactivity.

I made a [PR earlier for something similar](https://github.com/home-assistant/frontend/pull/28502), but that one removed the visibility toggle functionality, and did not implement the feature consistently over charts. This new PR solves these issues.

### Motivation
Users often want to quickly access details about the entity displayed in the graph (see more information about the device, see other entities from that device, change its location, change its name etc.). But as of now, it takes a lot of clicks and actions to do so.

Opening the entity in a modal is the behavior that uses are likely to expect when clicking on it, given that this is what happens on most other components in the dashboard. However, we want to keep the ability to toggle series visibility to give users an option to remove clutter when needed. This split interaction model provides both capabilities efficiently, with clear visual cues. 

## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml
type: history-graph
entities:
  - sensor.sun_next_dawn
title: History Graph
```

```yaml
type: statistics-graph
entities:
  - sensor.sun_next_dawn
title: Statistics Graph
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR is related to issue or discussion: https://github.com/orgs/home-assistant/discussions/2036

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

joepio committed:
feat: Split legend interaction (click vs label) with icons and hover effects

joepio commented:
Lmk if there is something that needs to be changed.

MindFreeze commented:
@laupalombi You removed the "Needs UX" label but I am still not clear on what the decision is here. Are we going with the checkbox and an icon after the label?
Also cc @marcinbauer85

MindFreeze commented on src/components/chart/state-history-chart-line.ts:
```diff
@@ -103,6 +103,8 @@ export class StateHistoryChartLine extends LitElement {
         .expandLegend=${this.expandLegend}
         .hideResetButton=${this.hideResetButton}
+        .externalHiddenState=${this.clickForMoreInfo}
+        @chart-legend-click=${this._handleLegendClick}
```
I don't like the names of these. External hidden state is already supported via `@dataset-hidden` so this is not needed for that exactly. You can see it in action in `hui-energy-devices-detail-graph-card`.
Events that start with `chart-` proxied echarts events. I think we shouldn't prefix the legend events so it is clear they are from HA, not echarts. Maybe `legend-label-click`?
I'll think about a recommendation.

karwosts reviewed (changes requested):
External statistics are showing as clickable link yet have no more-info, that ought to be disabled. 

Statistics graph with 1 entity and the expanded legend doesn't work right. 

The legend has min, mean,max, and if I click on `sensor.foo (max)`, it tries to load non-existant more-info with entity `sensor.foo-max`. 

The clickable bubble looks really dark in statistics graph, hard to see the checkbox:
Compare history:
<img width="114" height="34" alt="image" src="https://github.com/user-attachments/assets/e632554c-8c11-4ac2-8ae6-c453d35d9b55" />

statistics:
<img width="158" height="43" alt="image" src="https://github.com/user-attachments/assets/0cc2dcad-e387-445c-be96-4cff81b99dff" />

(this one may be a pre-existing issue, but just noticed especially it since the icon changed and I can barely see it)

marcinbauer85 commented:
Thanks for double checking @karwosts. The color contrast issue is not related to this PR.
Production:
<img width="392" height="79" alt="CleanShot 2026-01-14 at 09 50 27" src="https://github.com/user-attachments/assets/d6ee8d9c-f08e-487c-920a-1a76f40e85b7" />

PR:
<img width="361" height="63" alt="CleanShot 2026-01-14 at 09 50 41" src="https://github.com/user-attachments/assets/b114ec9e-5834-45fb-92ef-596e7f7f4b88" />

I would tackle chart colors in a separate PR.

MindFreeze committed:
Merge remote-tracking branch 'origin/dev' into pr/28517

# Conflicts:
#	src/components/chart/ha-chart-base.ts

MindFreeze committed:
Address review feedback on chart legend split

- Rename `externalHiddenState` to `clickLabelForMoreInfo` and the event
  `chart-legend-click` to `legend-label-click` (the `chart-` prefix is
  reserved for echarts-proxy events).
- Use proper translation keys
  `ui.components.history_charts.{toggle_visibility,show_more_info}`
  instead of `ui.common.*` which don't exist.
- Drop the redundant tooltip on the label when it would duplicate the
  icon's "Toggle visibility" tooltip.
- Resolve the legend dataset id back to a real `entity_id` before
  opening more-info: strip known climate-attribute suffixes for history
  charts (`-current_temperature`, `-target_temperature*`, `-heating`/
  `-cooling`/`-drying`/`-fan`), and skip `isExternalStatistic` ids for
  statistics charts. Only fire `hass-more-info` if `hass.states[id]`
  resolves.

MindFreeze committed:
Address followup review feedback on chart legend

- Restore stat-type suffix stripping in statistics chart. When the chart
  has a single entity, ha-chart-base falls back to raw series ids
  (`${statistic_id}-${type}`) for the legend, so clicking
  "sensor.foo (max)" otherwise tries to open `sensor.foo-max`.
- Cover humidifier multi-attribute datasets in the entity-id resolver
  alongside climate / water_heater (`-current_humidity`,
  `-target_humidity`, `-humidifying`, `-on`; `-drying` is already
  contributed by CLIMATE_MODE_CONFIGS). Rename the constant to
  ENTITY_DATASET_SUFFIXES to reflect the broader scope.
- Convert the two legend click targets to `<button type="button">` so
  Tab navigation, Enter/Space activation, and screen-reader semantics
  work. Add `aria-pressed` to the visibility toggle and a
  `:focus-visible` outline. Read the dataset id from `data-id` instead
  of `parentElement.id` so the handlers don't depend on DOM nesting.
- Scope the label hover underline to `.label-clickable` so charts that
  don't opt into more-info (network, sankey, sunburst, energy) keep the
  legacy non-underlined toggle behavior. Render the "Show more info"
  tooltip only when the click will actually fire.
- Drop the redundant physical `margin-right: 0` on the legend toggle;
  `margin-inline-end: 0` alone preserves the click-area extension on
  the start side in RTL.
